packages/gestalt/src/TypeaheadInputField.js

```diff
@@ -0,0 +1,169 @@
+// @flow strict
+
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import layout from './Layout.css';
+import styles from './TypeaheadInputField.css';
+import Box from './Box.js';
+import Icon from './Icon.js';
+import FormLabel from './FormLabel.js';
+
+type Props = {|
+  accessibilityLabel: string,
+  id: string,
+  onBlur?: ({ event: SyntheticEvent<HTMLInputElement> }) => void,
+  onChange: ({
+    value: string,
+    syntheticEvent: SyntheticEvent<HTMLInputElement>,
+  }) => void,
+  onClear?: () => void,
+  onFocus?: ({
+    value: string,
+    syntheticEvent: SyntheticEvent<HTMLInputElement>,
+  }) => void,
+  placeholder?: string,
+  size?: 'md' | 'lg',
+  value?: string,
+  label?: string,
+  forwardedRef?: React.Ref<'input'>,
+|};
+
+const InputField = ({
+  accessibilityLabel,
+  id,
+  label = id,
+  onBlur,
+  onChange,
+  onClear,
+  onFocus,
+  placeholder,
+  size = 'md',
+  value,
+  forwardedRef,
+}: Props) => {
+  const [hovered, setHovered] = React.useState<boolean>(false);
+
+  const handleChange = (event: SyntheticEvent<HTMLInputElement>) => {
+    onChange({
+      value: event.currentTarget.value,
+      syntheticEvent: event,
+    });
+  };
+
+  const handleClear = (event: SyntheticEvent<HTMLInputElement>) => {
+    onChange({ value: '', syntheticEvent: event });
+
+    if (onClear) onClear();
+  };
+  const handleMouseEnter = () => setHovered(true);
+
+  const handleMouseLeave = () => setHovered(false);
+
+  const handleFocus = (event: SyntheticEvent<HTMLInputElement>) => {
+    if (onFocus) {
+      onFocus({
+        value: event.currentTarget.value,
+        syntheticEvent: event,
+      });
+    }
+  };
+
+  const handleBlur = (event: SyntheticEvent<HTMLInputElement>) => {
+    if (onBlur) {
+      onBlur({ event });
+    }
+  };
+
+  const hasValue = value && value?.length > 0;
+
+  const className = classnames(styles.input, {
+    [layout.medium]: size === 'md',
+    [layout.large]: size === 'lg',
+    [styles.inputActive]: true,
+    [styles.inputHovered]: hovered,
+  });
+
+  const clearButtonSize = size === 'lg' ? 24 : 20;
+  const clearIconSize = size === 'lg' ? 12 : 10;
+
+  return (
+    <span>
+      {' '}
+      {label && <FormLabel id={id} label={label} />}
+      <Box
+        alignItems="center"
+        display="flex"
+        onMouseEnter={handleMouseEnter}
+        onMouseLeave={handleMouseLeave}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+        position="relative"
+      >
+        <input
+          ref={forwardedRef}
```

Comment:
@simkessy we probably want to add `autocomplete="off"` so it doesn't show the native autocomplete

![image](https://user-images.githubusercontent.com/127199/85082244-2bafe000-b183-11ea-9c4d-9fae3289bb25.png)